src/LinuxTracing/TracerImpl.cpp

```diff
@@ -72,21 +72,25 @@ TracerImpl::TracerImpl(
       listener_{listener} {
   CHECK(listener_ != nullptr);
 
-  if (unwinding_method_ != CaptureOptions::kUndefined) {
-    uint32_t stack_dump_size = capture_options.stack_dump_size();
-    if (stack_dump_size > kMaxStackSampleUserSize || stack_dump_size == 0) {
-      ERROR("Invalid sample stack dump size: %u; Reassigning to default: %u", stack_dump_size,
-            kMaxStackSampleUserSize);
-      stack_dump_size = kMaxStackSampleUserSize;
-    }
-    stack_dump_size_ = static_cast<uint16_t>(stack_dump_size);
-    std::optional<uint64_t> sampling_period_ns =
-        ComputeSamplingPeriodNs(capture_options.samples_per_second());
-    FAIL_IF(!sampling_period_ns.has_value(), "Invalid sampling rate: %.1f",
-            capture_options.samples_per_second());
-    sampling_period_ns_ = sampling_period_ns.value();
+  uint32_t stack_dump_size = capture_options.stack_dump_size();
+  if (stack_dump_size == std::numeric_limits<uint16_t>::max()) {
+    constexpr uint16_t kDefaultStackSampleUserSizeFramePointer = 512;
+    stack_dump_size = (unwinding_method_ == CaptureOptions::kDwarf)
+                          ? kMaxStackSampleUserSize
+                          : kDefaultStackSampleUserSizeFramePointer;
+    LOG("No sample stack dump size was set; Assigning to default: %u", stack_dump_size);
```

Comment:
A general and arguably worthless comment on this and the other logs. A capital letter after a semicolon is not really a thing in English, so even if this is not really a text I'd say prefer period+capital ("was set. Assigning") or semicolon+lowercase ("was set; assigning").